### PR TITLE
Avoiding to fail step if CodeCov finds an error

### PR DESCRIPTION
**Pull request by didierofrivia**
This PR avoids the CI step to fail if CodeCov encounters an error.

### CODE REVIEW

**Review by KevFan — APPROVED** [review]
This looks good to me ! 👍

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -63,7 +63,7 @@ jobs:
         uses: codecov/codecov-action@v3
         with:
           verbose: true
-          fail_ci_if_error: true
+          fail_ci_if_error: false
```
[the remark this answers is not in the document]
Comment (alexsnaps) [reply .github/workflows/rust.yml:66]: That or we add the `if: ${{ wherever_the_secret_is != '' }} condition to the step...

**Comment by alexsnaps** [comment]
sorry @didierofrivia , I copy/pasted from the other comment, but looks like this ain't a thing:
```
The workflow is not valid. .github/workflows/rust.yml (Line: 63, Col: 13): Unrecognized named-value: 'secrets'. Located at position 1 within expression: secrets.CODECOV_TOKEN != ''
```

based on a quick search, looks like this has to be split:

```yaml
  env:
    CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
  if: env.CODECOV_TOKEN != null
```

**Review by alexsnaps — APPROVED** [review]
It worked!

**Pull request merged**